src/App.tsx

```diff
@@ -55,6 +57,34 @@ loader.init().then((monaco) => {
   })
 })
 
+const getAppDataFromSearchParams = () => {
+  const urlParams = new URLSearchParams(window.location.search)
+  const encAppData = urlParams.get('appData')
+  if (!encAppData)
+    return {
+      schema: '',
+      values: [],
+    }
+
+  return appDataSchema.parse(
+    JSON.parse(LZString.decompressFromEncodedURIComponent(encAppData)),
```

Comment:
I honestly like nesting calls sometimes because:

1. you avoid naming variables
2. if the method's names are clear, the values are as well.

The param are on a newline because of prettier probably that defines a maxlength

Reply:
I would change prettier config then, anyway why would you avoid additional variables?
Sometimes naming stuff is hard, but is always important for different reasons. Quick example:
```
  const encodedAppData = urlParams.get('appData')
  if (!encodedAppData)
    return {
      schema: '',
      values: [],
    }

  const decodedAppData: string = LZString.decompressFromEncodedURIComponent(encodedAppData)
  const appData = JSON.parse(decodedAppData)

  return appDataSchema.parse(appData)
```
The code above **instantly** gives the readers an idea about what's happening and the different steps to extract the data from the URL.
Same is not true using nested calls (that's one of the reason for the addition of async await syntax to handle promises).

Reply:
I think you're referring to the callback hell aka pyramid of doom. Handling async code with callbacks is a totally different thing compared to this and has different consequences.

Btw yes, storing the values in vars is definitely a more debuggable and maintainable approach

Reply:
Yes, but are both related to nested stuff, let's avoid that.

Reply:
Already done